### PR TITLE
replaced file_exists with is_file

### PR DESCRIPTION
**Pull request by pine3ree**
a template file must be a file

### CODE REVIEW

File: wire/core/TemplateFile.php [hunk 1]
```diff
@@ -151,7 +151,7 @@ public function setGlobal($name, $value, $overwrite = true) {
 	public function ___render() {
 
 		if(!$this->filename) return '';
-		if(!file_exists($this->filename)) {
+		if(!is_file($this->filename)) {
```
Comment (harikt) [inline wire/core/TemplateFile.php]: Mostly I use `is_readable` http://php.net/is_readable

**Comment by ryancramerdesign** [comment]
I don't think this would work for symlink templates, which is something we want to support. Also in testing here, I've found file_exists to have a slight performance edge over is_file... whether that's just my environment or it translates elsewhere, I'm not sure.

**Comment by pine3ree** [comment]
Hi Ryan,

is_file should me faster than file_exists (and it 's ever been in my tests clearing stats cache in loops) and fails (obviously) for dirs. The ability of distinguish between dir and files 

I wasn't aware of your intent of supporting symlinked template files. 
If you only need to support files inside symlinked dirs, is_file will behave correctly. But you need to support linked templates directly, then you can close this PR, as is_file will fail.

kindly

**Comment by ryancramerdesign** [comment]
My own tests showed performance to be better with file_exists(), but I did some more research and it sounds like there's not a lot of consistency as to which one performs better. But people do consistently observe one to be performing better than the other, for one reason or another. It sounds like it may be affected by environmental factors. In any case, both use the same stat() function to do their work. I can't find any evidence independent of environmental factors showing that it's worth changing from one to another. In this case, I also think it's better to be more open ended with what allowed for template files, as I don't want to break compatibility for those that might be using symlink template files (as I sometimes do during development, and others may too).

**Comment by pine3ree** [comment]
Thanks for your tests Ryan. I'm closing the PR.
